env/node.fetcher/src/index.ts

```diff
@@ -23,6 +23,17 @@ export interface FetchNodeOptions {
 }
 
 export async function fetchNode (fetch: FetchFromRegistry, version: string, targetDir: string, opts: FetchNodeOptions) {
+  // all node versions up until now (Aug 7 2023) strictly follow `vX.Y.Z` format
+  // this code should be changed if the above condition ever ceases to apply
+  if (!opts.nodeMirrorBaseUrl && !/^[0-9]+\.[0-9]+\.[0-9]+$/.test(version)) {
```

Comment:
Why should it check for the existence of nodeMirrorBaseUrl?

Reply:
I don't think we should support non-standard URLs.

Reply:
OK. I will remove it.

```suggestion
  if (!/^[0-9]+\.[0-9]+\.[0-9]+$/.test(version)) {
```